### PR TITLE
Aes 76 verification

### PR DESCRIPTION
**Pull request by kBlazewicz**

### CODE REVIEW

File: src/main/kotlin/pl/edu/agh/server/domain/authentication/AuthenticationService.kt [hunk 1]
```diff
@@ -97,10 +110,52 @@ class AuthenticationService(
...
+
+        user.enabled = true
+        user.verificationToken = null
+        userRepository.save(user)
+    }
+
+    @Transactional
+    fun prepareForPasswordChange(request: ResetPasswordRequest) {
+        val user =
+            userRepository.findByEmail(request.email).orElseThrow { throw IllegalArgumentException("Invalid email") }
```
Comment (rafibz007) [inline src/main/kotlin/pl/edu/agh/server/domain/authentication/AuthenticationService.kt:153]: To wymagaloby raczej duzej zmiany i troche roboty, wiec mozesz smialo zignorowac, pisze bardziej jako ciekawostke.

Zwracanie bledu gdy uzytkownik poda niewlasciwego maila umozliwi enumeracje po mailach i ulatwi potencjalny przyszly brute force hasel. IMO powinno sie zwracac zawsze generyczna wiadomosc: "Jezeli podane dane sa prawidze to powinenes dostac maila". Dodatkowo czas odpowiedzi powinen byc nieodroznialny w przypadku poprawnego i niepoprawnego maila. W tym wypadku nawet jezeli wyjdziesz z funkcji gdy podano niewlasciwy email zamiast rzucic blad, to prawdopodobnie podanie w requescie dlugiego (200,300,400... znakow) hasla sprawi ze jego dalsze encodowanie zrobi duza roznice czasowa i dalej bedzie mozna enumerowac nazwy uzytkownikow sprawdzajac czas jaki dany request byl procesowany.

Problem enumeracji uztkownikow jest tez w innych funkcjach jak widze.

To by było moglo byc jako medium podatnosc zgloszone: https://docs.fluidattacks.com/criteria/vulnerabilities/026/

File: src/main/kotlin/pl/edu/agh/server/domain/user/User.kt [hunk 2]
```diff
@@ -64,6 +64,12 @@ class User(
     @OneToMany(mappedBy = "user", cascade = [CascadeType.ALL], orphanRemoval = true, fetch = FetchType.LAZY)
     var organizationUserRoles: MutableList<OrganizationUserRole> = mutableListOf()
 
+    var verificationToken: String? = null
```
Comment (rafibz007) [inline src/main/kotlin/pl/edu/agh/server/domain/user/User.kt:67]: Troche edge case, ale z tego co widze, jak uzytkownik sie zarejestruje i przed walidacja zmieni sobie haslo, to token mu sie nadpisze i juz nigdy nie bedzie mogl potwierdzic konta :D

File: src/test/resources/application.properties [hunk 3]
```diff
@@ -10,9 +10,15 @@ spring.servlet.multipart.max-request-size=10MB
 file.upload-dir=${FILE_UPLOAD_DIRECTORY:./images}
 file.ddl-auto=create-drop
 configuration.mock-data=false
+configuration.email.verification=true
+application.email.username=agheventscheduler@gmail.com
+application.email.password=eogbhrohgvjwgcmf
```
Comment (rafibz007) [inline src/test/resources/application.properties]: Czy to sa "produkcyjne" kredki do poczty na publicznym repozytorium? :) 

Jak tak, to zmien to zeby bralo z enva, daj nam to haslo messengerem i to moze lepiej zmien. Znajac ludzi i ich skanery, to to moze juz latac po necie :(
